test/NormalModule.test.js

```diff
@@ -0,0 +1,410 @@
+			beforeEach(function() {
+				normalModule._source = null;
+			});
+			it("calls hash function with \"null\"", function() {
+				normalModule.updateHashWithSource(hash);
+				hashSpy.callCount.should.eql(1);
+				hashSpy.args[0][0].should.eql("null");
+			});
+		});
+		describe("without the module having source", function() {
+			let expectedSource = "wurst suppe";
+			beforeEach(function() {
+				normalModule._source = new RawSource(expectedSource);
+			});
+			it("calls hash function with \"source\" and then the actual source of the module", function() {
+				normalModule.updateHashWithSource(hash);
+				hashSpy.callCount.should.eql(2);
+				hashSpy.args[0][0].should.eql("source");
+				hashSpy.args[1][0].should.eql(expectedSource);
+			});
+		});
+	});
+	describe("#needRebuild", function() {
+		let fileTimestamps;
+		let contextTimestamps;
+		let fileDependencies;
+		let contextDependencies;
+		let fileA;
+		let fileB;
+
+		function setDeps(
+			fileDependencies,
+			contextDependencies) {
+			normalModule.fileDependencies = fileDependencies;
+			normalModule.contextDependencies = contextDependencies;
+		}
+
+		beforeEach(function() {
+			fileA = "fileA";
+			fileB = "fileB";
+			fileDependencies = [fileA, fileB];
+			contextDependencies = [fileA, fileB];
+			fileTimestamps = {
+				[fileA]: 1,
+				[fileB]: 1,
+			};
+			contextTimestamps = {
+				[fileA]: 1,
+				[fileB]: 1,
+			};
+			normalModule.buildTimestamp = 2;
+			setDeps(fileDependencies, contextDependencies);
+		});
+		describe("given all timestamps are older than the buildTimestamp", function() {
+			it("returns false", function() {
+				normalModule.needRebuild(fileTimestamps, contextTimestamps).should.eql(false);
+			});
+		});
+		describe("given a file timestamp is newer than the buildTimestamp", function() {
+			beforeEach(function() {
+				fileTimestamps[fileA] = 3;
+			});
+			it("returns true", function() {
+				normalModule.needRebuild(fileTimestamps, contextTimestamps).should.eql(true);
+			});
+		});
+		describe("given a no file timestamp exists", function() {
+			beforeEach(function() {
+				fileTimestamps = {};
+			});
+			it("returns true", function() {
+				normalModule.needRebuild(fileTimestamps, contextTimestamps).should.eql(true);
+			});
+		});
+		describe("given a context timestamp is newer than the buildTimestamp", function() {
+			beforeEach(function() {
+				contextTimestamps[fileA] = 3;
+			});
+			it("returns true", function() {
+				normalModule.needRebuild(fileTimestamps, contextTimestamps).should.eql(true);
+			});
+		});
+		describe("given a no context timestamp exists", function() {
+			beforeEach(function() {
+				contextTimestamps = {};
+			});
+			it("returns true", function() {
+				normalModule.needRebuild(fileTimestamps, contextTimestamps).should.eql(true);
+			});
+		});
+	});
+	describe("#splitVariablesInUniqueNamedChunks", function() {
+		let variables;
+		beforeEach(function() {
+			variables = [{
+					name: "foo"
+				},
```

Comment:
The array styling looks weird.

The following should be accepted by the linter:

``` js
variables = [{
  name: "foo"
}, {
  name: "bar"
}]
```